src/components/RichTextEditor/TextStyleDropdown/DropdownTrigger.tsx

```diff
@@ -1,45 +1,86 @@
 /* (c) Copyright Frontify Ltd., all rights reserved. */
 
-import { IconCaretDown } from '@foundation/Icon';
-import { getParent, getPreventDefaultHandler, usePlateEditorState } from '@udecode/plate';
-import { merge } from '@utilities/merge';
-import React from 'react';
-import { TextStyles, textStyleTitles } from '../utils/getTextStyles';
+import { IconCaretDown } from "@foundation/Icon";
+import {
+    PlateEditor,
+    getNodes,
+    getPreventDefaultHandler,
+    isElement,
+    isType,
+    usePlateEditorState,
+} from "@udecode/plate";
+import { merge } from "@utilities/merge";
+import React from "react";
+import { TextStyles, textStyleTitles } from "../utils/getTextStyles";
 
 type DropdownTriggerProps = {
     editorId?: string;
     open: boolean;
 };
+const DEFAULT_TEXTSTYLE_VALUE = "Mixed";
+
+enum ListType {
+    UL = "ul",
+    OL = "ol",
+    CHECKLIST_ITEM = "checkbox_item",
+}
+
+const listTitle: Record<ListType, string> = {
+    [ListType.UL]: "Bullet List",
+    [ListType.OL]: "List",
+    [ListType.CHECKLIST_ITEM]: "Checklist",
+};
+
+type AvailableTextStyles = ListType & TextStyles;
+
+const availableTextStyleTitles: Record<AvailableTextStyles, string> = { ...listTitle, ...textStyleTitles };
+
+const getSelectedTextStyles: (editor: PlateEditor) => AvailableTextStyles[] = (editor) => {
```

Comment:
```ts
const getSelectedTextStyles: (editor: PlateEditor) => AvailableTextStyles[] = (editor) => {
    if (!editor.selection) {
        return [];
    }
        
    return Array.from(
        getNodes(editor, {
            unhang: true,
            at: editor.selection,
            match: (node) => isElement(node) && isType(editor, node, Object.values({ ...TextStyles, ...ListType })),
        }),
    ).reduce<AvailableTextStyles[]>((types, current) => {
        const type = current[0].type as AvailableTextStyles;
        if (!types.includes(type)) {
            types.push(type);
        }
         return types;
    }, []);
};
```

Reply:
could you create a custom hook from this and get the editor via plate hook, so this can be reused?